src/wallet/redux/reducers/funding.ts

```diff
@@ -184,8 +194,23 @@ const bWaitForDepositToBeSentToMetaMaskReducer = (state: states.BWaitForDepositT
 
 const bSubmitDepositInMetaMaskReducer = (state: states.BSubmitDepositInMetaMask, action: actions.WalletAction) => {
   switch (action.type) {
+    // This case should not happen in theory, but it does in practice.
```

Comment:
Its crazy that this can happen before transaction submitted

Reply:
@lalexgap This might not be a realistic scenario when using Ropsten or the mainchain, but it does happen locally with ganache.